### PR TITLE
Fixes #3537: Add "Update image" button

### LINKED ISSUE #3537: Add a "Update image" button in the summary tab
Add a "Update image" button in the summary tab above the "Take another picture" button and rename the "Take another picture" button to "Take extra photos of the packaging"
The "Update image" should replace the front image for the current language.

### PR DESCRIPTION
**Pull request by WildOrangutan**
Is this what you had in mind? Hopefully I'm not missing something here.

#3537

### CODE REVIEW

File: app/src/main/res/layout/fragment_summary_product.xml [hunk 1]
```diff
@@ -628,14 +628,26 @@
...
 
+                <Button
+                    android:id="@+id/buttonNewFrontImage"
+                    style="@style/ButtonBorder"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_horizontal"
+                    android:layout_marginStart="@dimen/spacing_normal"
+                    android:layout_marginEnd="@dimen/spacing_normal"
+                    android:text="@string/update_image"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_summary_product.xml:639]: send_updated_image ?

Comment (teolemon) [reply app/src/main/res/layout/fragment_summary_product.xml:639]: <string name="send_updated_image">Send an Updated Image</string>
    <string name="update_image">Update image</string>
<string name="take_extra_packaging_pictures">Take extra photos of packaging</string>

Comment (teolemon) [reply app/src/main/res/layout/fragment_summary_product.xml:639]: it's the same meaning, so no big deal

**Comment by WildOrangutan** [comment]
I think I'm still missing uploading of new front image in the Fragment. Give me some time to also address that.

File: app/src/main/res/layout/fragment_summary_product.xml (hunk 1 above)
Comment (WildOrangutan) [reply app/src/main/res/layout/fragment_summary_product.xml:639]: Yeah, was confused a bit if would be ok. No problem, I will revert adding of this new resource.

**Comment by WildOrangutan** [comment]
I think this should suffice. There are also some issues with refreshing front image, after it changes, but I'd say this is a different problem.

**Comment by teolemon** [comment]
It works for me 👍

**Review by teolemon — APPROVED** [review]
This looks and behaves as expected. @VaiTon for a code-side review ?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java [hunk 2]
```diff
@@ -231,6 +220,29 @@ public void onError(Throwable error) {
         refreshView(productState);
     }
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java:222]: Could you add JavaDocs?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java [hunk 3]
```diff
@@ -231,6 +220,29 @@ public void onError(Throwable error) {
...
+            public void onComplete() {
+                onImageListenerComplete();
+            }
+            @Override
+            public void onError(Throwable error) {
+                onImageListenerError(error);
+            }
+        }));
+    }
+
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java:236]: Could you add JavaDocs?

**Review by VaiTon — APPROVED** [review]
Other than Javadocs, LGTM!

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java (hunk 2 above)
Comment (WildOrangutan) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java:222]: Will do

**Pull request merged**